ivy/functional/backends/torch/linear_algebra.py

```diff
@@ -147,7 +147,15 @@ def matrix_rank(
     out: Optional[torch.Tensor] = None,
 ) -> torch.Tensor:
     # ToDo: add support for default rtol value here, for the case where None is provided
-    ret = torch.linalg.matrix_rank(x, rtol=rtol, out=out)
+    if rtol is None and out is None:
```

Comment:
can you please add a comment explaining why each of these conditions are needed. Seems like `ret = torch.linalg.matrix_rank(x, rtol=rtol, out=out)` should work? Are you testing with the latest Dockerfile?

Reply:
I am not using Dockers since its quite a heavy setup which my laptop can't sustain. Nonetheless, after rechecking, I did realize that I was using an older version of torch which is why I had to handle the `NoneType `arguments explicitly. But yeah, now I have removed that unnecessary logic from the torch backend